com.hazelcast.connector.Hz3EnrichmentTest [HZ-1925]

_master_ (commit 4f848dc1a2f740dcda30c7f9ce36aec5ed9ce52e)

Failed on Windows - MicrosoftJDK11: https://jenkins.hazelcast.com/job/Hazelcast-master-Windows-MicrosoftJDK11/43/testReport/junit/com.hazelcast.connector/Hz3EnrichmentTest/testMapUsingReplicatedMap/
<details><summary>Stacktrace:</summary>

```
java.lang.RuntimeException: Problem in invoking Maven dependency:get OS_TEST_JAR:3.12.13
	at com.hazelcast.test.starter.HazelcastVersionLocator.downloadArtifact(HazelcastVersionLocator.java:96)
	at com.hazelcast.test.starter.HazelcastVersionLocator.locateArtifact(HazelcastVersionLocator.java:81)
	at com.hazelcast.test.starter.HazelcastVersionLocator.locateVersion(HazelcastVersionLocator.java:67)
	at com.hazelcast.test.starter.hz3.Hazelcast3Proxy.newHazelcastInstance(Hazelcast3Proxy.java:61)
	at com.hazelcast.test.starter.hz3.Hazelcast3Starter.newHazelcastInstance(Hazelcast3Starter.java:34)
	at com.hazelcast.connector.BaseHz3Test.setUp(BaseHz3Test.java:107)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:59)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:56)
	at org.junit.internal.runners.statements.RunBefores.invokeMethod(RunBefores.java:33)
	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:24)
	at org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:27)
	at com.hazelcast.test.DumpBuildInfoOnFailureRule$1.evaluate(DumpBuildInfoOnFailureRule.java:37)
	at com.hazelcast.test.jitter.JitterRule$1.evaluate(JitterRule.java:104)
	at com.hazelcast.test.metrics.MetricsRule$1.evaluate(MetricsRule.java:63)
	at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
	at org.junit.runners.BlockJUnit4ClassRunner$1.evaluate(BlockJUnit4ClassRunner.java:100)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:366)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:103)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:63)
	at org.junit.runners.ParentRunner$4.run(ParentRunner.java:331)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:79)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:329)
	at org.junit.runners.ParentRunner.access$100(ParentRunner.java:66)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:293)
	at org.junit.rules.TestWatcher$1.evaluate(TestWatcher.java:61)
	at org.junit.rules.RunRules.evaluate(RunRules.java:20)
	at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:413)
	at org.junit.runners.Suite.runChild(Suite.java:128)
	at org.junit.runners.Suite.runChild(Suite.java:27)
	at org.junit.runners.ParentRunner$4.run(ParentRunner.java:331)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:79)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:329)
	at org.junit.runners.ParentRunner.access$100(ParentRunner.java:66)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:293)
	at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:413)
	at org.apache.maven.surefire.junitcore.JUnitCore.run(JUnitCore.java:55)
	at org.apache.maven.surefire.junitcore.JUnitCoreWrapper.createRequestAndRun(JUnitCoreWrapper.java:137)
	at org.apache.maven.surefire.junitcore.JUnitCoreWrapper.executeEager(JUnitCoreWrapper.java:107)
	at org.apache.maven.surefire.junitcore.JUnitCoreWrapper.execute(JUnitCoreWrapper.java:83)
	at org.apache.maven.surefire.junitcore.JUnitCoreWrapper.execute(JUnitCoreWrapper.java:75)
	at org.apache.maven.surefire.junitcore.JUnitCoreProvider.invoke(JUnitCoreProvider.java:157)
	at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:428)
	at org.apache.maven.surefire.booter.ForkedBooter.execute(ForkedBooter.java:162)
	at org.apache.maven.surefire.booter.ForkedBooter.run(ForkedBooter.java:562)
	at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:548)
Caused by: java.io.IOException: Cannot run program "mvn": CreateProcess error=2, The system cannot find the file specified
	at java.base/java.lang.ProcessBuilder.start(ProcessBuilder.java:1128)
	at java.base/java.lang.ProcessBuilder.start(ProcessBuilder.java:1071)
	at com.hazelcast.test.starter.HazelcastVersionLocator.downloadArtifact(HazelcastVersionLocator.java:91)
	... 51 more
Caused by: java.io.IOException: CreateProcess error=2, The system cannot find the file specified
	at java.base/java.lang.ProcessImpl.create(Native Method)
	at java.base/java.lang.ProcessImpl.<init>(ProcessImpl.java:487)
	at java.base/java.lang.ProcessImpl.start(ProcessImpl.java:154)
	at java.base/java.lang.ProcessBuilder.start(ProcessBuilder.java:1107)
	... 53 more

```
</details>


<details><summary>Standard output:</summary>

```
BuildInfo right after testMapUsingReplicatedMap(com.hazelcast.connector.Hz3EnrichmentTest): BuildInfo{version='5.3.0-SNAPSHOT', build='20221215', buildNumber=20221215, revision=4f848dc, enterprise=false, serializationVersion=1}
Hiccups measured while running test 'testMapUsingReplicatedMap(com.hazelcast.connector.Hz3EnrichmentTest):'
20:46:55, accumulated pauses: 1 ms, max pause: 0 ms, pauses over 1000 ms: 0


No metrics recorded during the test

```
</details>

Standard output can be found here - https://s3.console.aws.amazon.com/s3/buckets/j-artifacts/Hazelcast-master-Windows-MicrosoftJDK11/43/

The same error was also in other tests from this build. 
It fails because instead of "mvn" we should use "mvn.cmd"  on Windows OS

--- a/hazelcast/src/test/java/com/hazelcast/test/starter/HazelcastVersionLocator.java
+++ b/hazelcast/src/test/java/com/hazelcast/test/starter/HazelcastVersionLocator.java
@@ -27,6 +27,7 @@ import java.util.HashMap;
 import java.util.Map;
 import java.util.concurrent.TimeUnit;
 
+import static com.hazelcast.internal.util.OsHelper.isWindows;
 import static com.hazelcast.internal.util.Preconditions.checkState;
 import static com.hazelcast.test.JenkinsDetector.isOnJenkins;
 import static java.io.File.separator;
@@ -109,7 +110,8 @@ public class HazelcastVersionLocator {
     }
 
     private static String buildMavenCommand(Artifact artifact, String version) {
-        return "mvn dependency:get -Dartifact=com.hazelcast:"
+        String mvn = isWindows() ? "mvn.cmd" : "mvn";
+        return mvn + " dependency:get -Dartifact=com.hazelcast:"
                 + artifact.mavenProject + ":" + version
                 + (artifact.test ? ":jar:tests" : "")
                 + (artifact.enterprise ? " -DremoteRepositories=https://repository.hazelcast.com/release" : "");

